soc/arm/microchip_mec/mec172x/reg/mec172x_regs.h

```diff
@@ -0,0 +1,1119 @@
+/*
+ * Copyright (c) 2009-2018 Arm Limited. All rights reserved.
+ * Copyright (c) 2021 Microchip Technology Inc. and its subsidiaries.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef MEC172X_REGS_H
+#define MEC172X_REGS_H
+
+#include <stddef.h>
+#include <stdint.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/*
+ * MEC172x includes the ARM single precicision FPU and the ARM MPU with
+ * eight regions.
+ */
+#define __MPU_PRESENT		1
+#define __FPU_PRESENT		1
+
+struct btmr_regs {
+	volatile uint32_t CNT;
+	volatile uint32_t PRLD;
+	volatile uint8_t STS;
+	uint8_t RSVDC[3];
+	volatile uint8_t IEN;
+	uint8_t RSVDD[3];
+	volatile uint32_t CTRL;
+};
+
+/** @brief GIRQ registers. Total size = 20(0x14) bytes */
+struct girq_regs {
+	volatile uint32_t SRC;
+	volatile uint32_t EN_SET;
+	volatile uint32_t RESULT;
+	volatile uint32_t EN_CLR;
+	uint32_t RSVD1[1];
+};
+
+/** @brief ECIA registers with each GIRQ elucidated */
+struct ecia_regs {
+	struct girq_regs GIRQ08;
+	struct girq_regs GIRQ09;
+	struct girq_regs GIRQ10;
+	struct girq_regs GIRQ11;
+	struct girq_regs GIRQ12;
+	struct girq_regs GIRQ13;
+	struct girq_regs GIRQ14;
+	struct girq_regs GIRQ15;
+	struct girq_regs GIRQ16;
+	struct girq_regs GIRQ17;
+	struct girq_regs GIRQ18;
+	struct girq_regs GIRQ19;
+	struct girq_regs GIRQ20;
+	struct girq_regs GIRQ21;
+	struct girq_regs GIRQ22;
+	struct girq_regs GIRQ23;
+	struct girq_regs GIRQ24;
+	struct girq_regs GIRQ25;
+	struct girq_regs GIRQ26;
+	uint8_t RSVD2[(0x0200u - 0x017Cu)];
```

Comment:
lower-case hex

also, I really hate the u suffix. What a shame if this is required by Zephyr

Reply:
Hi @sjg20 Done.  Created new macros to cast numbers to uint32_t for address calculations. Pass literals without u suffix to macro. Hopefully the cast in the macro will satisfy MISRA rules.